designer_v2/lib/features/dashboard/studies_table.dart

```diff
@@ -14,60 +17,146 @@ class StudiesTable extends StatelessWidget {
+          label: tr.studies_list_header_title,
+          columnWidth: const MaxColumnWidth(FixedColumnWidth(200), FlexColumnWidth(2.4)),
+          sortable: true,
+        ),
         StandardTableColumn(
-            label: tr.studies_list_header_participation,
-            columnWidth: const MaxColumnWidth(FixedColumnWidth(130), IntrinsicColumnWidth())),
+          label: tr.studies_list_header_status,
+          columnWidth: const MaxColumnWidth(FixedColumnWidth(90), IntrinsicColumnWidth()),
+          sortable: true,
+        ),
+        StandardTableColumn(
+          label: tr.studies_list_header_participation,
+          columnWidth: const MaxColumnWidth(FixedColumnWidth(130), IntrinsicColumnWidth()),
+          sortable: true,
+        ),
         StandardTableColumn(
           label: tr.studies_list_header_created_at,
           columnWidth: const FlexColumnWidth(1),
+          sortable: true,
         ),
         StandardTableColumn(
           label: tr.studies_list_header_participants_enrolled,
-          columnWidth: const FlexColumnWidth(0.5),
+          columnWidth: MaxColumnWidth(FixedColumnWidth(statsCellWidth), const IntrinsicColumnWidth()),
+          sortable: true,
         ),
         StandardTableColumn(
           label: tr.studies_list_header_participants_active,
-          columnWidth: const FlexColumnWidth(0.5),
+          columnWidth: MaxColumnWidth(FixedColumnWidth(statsCellWidth), const IntrinsicColumnWidth()),
+          sortable: true,
         ),
         StandardTableColumn(
           label: tr.studies_list_header_participants_completed,
-          columnWidth: const FlexColumnWidth(0.5),
+          columnWidth: MaxColumnWidth(FixedColumnWidth(statsCellWidth), const IntrinsicColumnWidth()),
+          sortable: true,
         ),
       ],
       onSelectItem: onSelect,
       trailingActionsAt: (item, _) => getActions(item),
       buildCellsAt: _buildRow,
+      sortColumnPredicates: _sortColumns,
+      pinnedPredicates: pinnedPredicates,
       emptyWidget: emptyWidget,
     );
   }
 
+  int Function(Study a, Study b) get pinnedPredicates {
+    return (Study a, Study b) {
+      if (pinnedStudies.contains(a.id)) {
+        return -1;
+      } else if (pinnedStudies.contains(b.id)) {
+        return 1;
+      }
+      return 0;
+    };
+  }
+
+  List<int Function(Study a, Study b)?> get _sortColumns {
+    final predicates = [
+      (Study a, Study b) => 0, // do not sort pin icon
+      (Study a, Study b) => a.title!.compareTo(b.title!),
+      (Study a, Study b) => a.status.index.compareTo(b.status.index),
+      (Study a, Study b) => a.participation.index.compareTo(b.participation.index),
+      (Study a, Study b) => a.createdAt!.compareTo(b.createdAt!),
+      (Study a, Study b) => a.participantCount.compareTo(b.participantCount),
+      (Study a, Study b) => a.activeSubjectCount.compareTo(b.activeSubjectCount),
+      (Study a, Study b) => a.endedCount.compareTo(b.endedCount),
```

Comment:
Not sure if this is the right place for this or if it happens somewhere else/is already taken care of:

Since Dart's `sort` is not stable for equally comparing items, maybe we should sort by IDs as a final fallback so nobody gets confused that the order can randomly change for studies that compare to be equal?

Reply:
This did not occur during my tests, but I think it's worth implementing the addition. 👍